src/js/actions/superselect.js

```diff
@@ -188,19 +188,37 @@ define(function (require, exports) {
     };
 
     /**
-     * Gets the non group layers that have one of the passed in IDs
+     * Gets the artboards and leaf layers that have one of the passed in IDs
      * 
      * @param  {LayerStructure} layerTree
      * @param  {Object.<{number: boolean}>} layerMap       
      * @return {Immutable.Iterable.<Layer>}
      */
-    var _getLeafLayersWithID = function (layerTree, layerMap) {
-        return layerTree.leaves.filter(function (layer) {
+    var _getDirectAccessLayersWithID = function (layerTree, layerMap) {
+        var artboards = layerTree.top.filter(function (layer) {
```

Comment:
I guess maybe we've moved beyond the "cache everything" fad of the earlier this year, but seems like "allArtboards" would be a reasonable layerstructure property.